Remove dockstore-webservice dependency

kathy-t opened the pull request:
**Description**
Companion webservice PR https://github.com/dockstore/dockstore/pull/5562

This PR removes the dockstore-webservice dependency by only using things from dockstore-common. Note that the metrics-aggegator still uses the dockstore-webservice with the test scope.

I also removed all javax dependencies because I didn't see any imports for them. Looks like javax imports were deleted in https://github.com/dockstore/dockstore-support/pull/468 when we removed the jenkins functionality. This will remove the need for https://github.com/dockstore/dockstore/issues/5527.

**Review Instructions**
Tests should pass and the metricsaggregator and tooltester should still be functional. Can try running a command for both the metrics aggregator and tooltester to make sure the JAR works. For the metrics aggregator, could submit validation data using a simple CSV file containing data for one workflow. For the tooltester, could upload metrics.

**Issue**
dockstore/dockstore#5528
https://ucsc-cgl.atlassian.net/browse/DOCK-2404

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `mvn clean install` in the project that you have modified (until https://ucsc-cgl.atlassian.net/browse/SEAB-5300 adds multi-module support properly)
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] If you are changing dependencies, check with dependabot to ensure you are not introducing new high/critical vulnerabilities
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.

denis-yuen commented on THIRD-PARTY-LICENSES.txt:
```diff
@@ -1,14 +1,14 @@
 
-Lists of 411 third-party dependencies.
+Lists of 399 third-party dependencies.
```
12 down, I see things like `dockstore-integration-testing` and `dropwizard-testing` which makes me think a lot of what I thought would be removed is staying around from a licenses POV due to that test scope. 

What is the difference in jar size (or maven shade plugin warning)?

denis-yuen replied:
Also relevant to https://github.com/dockstore/dockstore/issues/5549

denis-yuen replied:
A develop webservice is 159M, I think that drop is significant.

denis-yuen commented on pom.xml:
```diff
@@ -598,9 +541,9 @@
-                                        <!-- used by toolbackup <exclude>javax.annotation:javax.annotation-api</exclude> -->
-                                        <!-- still used by tooltester <exclude>javax.activation:activation</exclude> -->
-                                        <!-- still used by tooltester <exclude>javax.ws.rs:javax.ws.rs-api</exclude> -->
+                                        <exclude>javax.annotation:javax.annotation-api</exclude>
```
Nice

denis-yuen commented on toolbackup/pom.xml:
```diff
@@ -170,12 +170,14 @@
         <dependency>
             <groupId>com.amazonaws</groupId>
             <artifactId>aws-java-sdk-core</artifactId>
+            <version>1.11.83</version>
```
Odd, these two should be in our bom via the AWS bom 
https://github.com/dockstore/dockstore/blob/develop/bom-internal/pom.xml#L337-L342

kathy-t replied:
Looking at the [package names](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/migration-whats-different.html), our bom has the AWS version 2 SDK, but toolbackup uses version 1

denis-yuen reviewed (commented):
Likely ok, but will return after reviewing webservice PR

denis-yuen reviewed (approved):
After the other one of course

kathy-t commented:
I found a problem when running the tooltester and metrics aggregator so I removed reviewers and made this PR a draft. Will re-request reviews when it's ready again

kathy-t commented:
When running the metrics aggregator and tool tester locally, they failed. Turns out they need the `org.glassfish.jersey.inject:jersey-hk2` dependency to run. Disclaimer: I only tested the metrics aggregator and tooltester commands, not the toolbackup, but it seems like we currently don't use it.

I ran into a null pointer exception when running workflows on AGC using the tooltester so I added a null check.

denis-yuen commented on metricsaggregator/pom.xml:
```diff
@@ -89,35 +89,20 @@
-                    <artifactId>jackson-module-jaxb-annotations</artifactId>
-                </exclusion>
-            </exclusions>
+            <scope>test</scope>
```
:partying_face:

denis-yuen commented on toolbackup/pom.xml:
```diff
@@ -170,12 +170,14 @@
         <dependency>
             <groupId>com.amazonaws</groupId>
             <artifactId>aws-java-sdk-s3</artifactId>
+            <version>1.11.83</version>
```
Can use a property for these two to hint to a future developer that either these two are upgraded at once or more likely, deleted at the same time

denis-yuen commented on tooltester/pom.xml:
```diff
@@ -169,37 +166,11 @@
                 </exclusion>
             </exclusions>
         </dependency>
-        <dependency>
```
:+1:

denis-yuen reviewed (approved):
Thanks for checking for runtime errors, those can be pretty annoying to track down since they don't show up in tests

kathy-t commented:
Updated the dockstore-core dependency to a tag. Build failed because of the new cwlavro version in the webservice, so I updated the pom to use the cwlavro dependencies from the bom
